src/Akka.Persistence.Azure.Tests/AzureTableJournalEscapePersistentIdSpec.cs

```diff
@@ -0,0 +1,32 @@
+using System;
+using System.Reflection;
+using Akka.Actor;
+using Akka.Configuration;
+using Akka.Persistence.Azure.TestHelpers;
+using Akka.Persistence.TCK;
+using Akka.Persistence.TCK.Journal;
+using Akka.TestKit;
+using Xunit;
+using Xunit.Abstractions;
+using static Akka.Persistence.Azure.Tests.Helper.AzureStorageConfigHelper;
+
+namespace Akka.Persistence.Azure.Tests
+{
+    public class AzureTableJournalEscapePersistentIdSpec : AzureTableJournalSpec, IClassFixture<WindowsAzureStorageEmulatorFixture>
+    {
+        public AzureTableJournalEscapePersistentIdSpec(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        /// <inheritdoc />
+        protected override void PreparePersistenceId(string pid)
+        {
+            base.PreparePersistenceId(pid);
+            
+            // Before storage is initialized, let's set Pid to the value that needs to be encoded
+            var persistenceIdUsedForTests = typeof(PluginSpec).GetField($"<{nameof(Pid)}>k__BackingField", BindingFlags.Instance | BindingFlags.NonPublic);
+            var currentValue = persistenceIdUsedForTests.GetValue(this).ToString();
+            persistenceIdUsedForTests.SetValue(this, $"some/path/to/encode/{currentValue}");
```

Comment:
Little bit hacky, but allows to overwrite `Pid` property with invalid partition key - did not find better way after long investigation.
Here we do usual tests, but with paths that need to be escaped